### PR TITLE
Complete Budget App

### PR DESCRIPTION
**Pull request by Cluab**
#### Design
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

**Comment by Cluab** [comment]
# Presentation Video [Link](https://www.youtube.com/watch?v=syV1cW5HCvg)

**Comment by ShadyShawkat** [comment]
## Hi @Cluab, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has an unauthorized collaborator. Please make sure to only commit code using the GitHub account you registered in Microverse with. Also, there're 5,000+ files changed in this PR which makes it nearly impossible to see and review your code.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Cluab** [comment]
Hello @ShadyShawkat,

I noticed that the file changes may have caught your attention. I accidentally pushed the vendor file that contained the Gem folders, and for best practice, I had to remove them. Regarding the unauthorized collaborator issue, I encountered an issue previously where my commits were not linked to my GitHub account, causing me to lose over 100 commits. To resolve this, I contacted GitHub support, and they suggested that I change my git commits config to match my email, which I did by running the command `git config --global user.email "cluab11222@gmail.com"`. However, I am not sure how this has impacted my authorization status.

If you have any suggestions on how I can resolve the unauthorized collaborator issue, I would appreciate your input.

Thank you.
Screenshot from 2023-03-22 17-39-03

**Comment by ShadyShawkat** [comment]
Hello @Cluab, I'd advise you to contact Microverse's support about this issue. As a code reviewer, I'm led to invalidated pull requests that have an unauthorized collaborator which is an account that isn't registered in Microverse.

**Comment by enniopo** [comment]
@ShadyShawkat, thanks for catching this. Well done. 

For the next CR: **please proceed with the review**. 

@Cluab please submit another code review request through the dashboard.

**Comment by addod19** [comment]
### Hi @Cluab, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because you did not follow the Git-flow workflow thus creating a PR from lets say `auth` branch into the `development` branch

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Cluab** [comment]
Hello, @addod19 thanks for the review but I'm still not understanding the issue, I'm following the correct git flow where I completed the entire application and currently waiting for your approval to merge it with the main branch, the scenario that your talking about is if I was working in a specific feature and wanting to merge it with development branch.
I'm currently working on the rails capstone project

**Comment by Cluab** [comment]
@addod19 and as you can see over here with one of my peer's repo they completed the project and created a pull request to merge with the main branch [Link](https://github.com/V-Blaze/Budget_App/pull/6)

**Comment by Cluab** [comment]
* Message to the next reviewer if you feel that my work in this pull request is considered an invalid Code Review Request please get in touch with me on slack (Slack username: Ibrahim Suhail Al-Yousefi).

File: README.md [hunk 1]
```diff
@@ -1,24 +1,129 @@
...
+The Recipe App was built with:
 
-* Database initialization
+- Ruby On Rails
+- PostgreSQL
+
+### Live Demo
+
+You can view a live demo of the Recipe App at [Link](https://budget-app-ftb2.onrender.com/).
+
```
Comment (emmanuellekamwa) [inline README.md:40]: - [x] One of the requirements of this project is to record a 5-minutes loom video where you present your project. Please make sure you add it.

**Comment by addod19** [comment]
> Hello, @addod19 thanks for the review but I'm still not understanding the issue, I'm following the correct git flow where I completed the entire application and currently waiting for your approval to merge it with the main branch, the scenario that your talking about is if I was working in a specific feature and wanting to merge it with development branch. I'm currently working on the rails capstone project

Hi @Cluab, sorry for the misunderstanding. But in this project, it is a requirement to use Git-flow.
Screenshot from 2023-03-23 12-38-46

**Comment by addod19** [comment]
> @addod19 and as you can see over here with one of my peer's repo they completed the project and created a pull request to merge with the main branch [Link](https://github.com/V-Blaze/Budget_App/pull/6)

@Cluab, yes but some of us make mistakes

**Comment by Cluab** [comment]
> Hi @Cluab, sorry for the misunderstanding. But in this project, it is a requirement to use Git-flow. Screenshot from 2023-03-23 12-38-46

* @addod19 I did what is required and that is opening a pull request from dev in my case development to the main branch after adding all the necessary approved features to it.
* Can you please clarify the issue in more depth so that I can understand how I'm not following the 
 git-flow

**Comment by Cluab** [comment]
@addod19  as you can see over here it says from `development` to `main`
---------------------
Screenshot from 2023-03-23 14-47-19

**Review by emmanuellekamwa — CHANGES_REQUESTED** [review]
**STATUS: CHANGES REQUESTED ⛔🛑**
Hello @Cluab 👋
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Well-documented README.
- All linters are passing all the checks.
- Descriptive PR.

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Cluab** [comment]
Hello @emmanuellekamwa I have submitted the presentation video at the top of the pull request you can check it out

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Cluab,

✔️ Nice design

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**